LykkeApi2/Controllers/OrdersController.cs

```diff
@@ -122,32 +118,35 @@ public async Task<IActionResult> CancelLimitOrder(string orderId)
                 return BadRequest("Session confirmation is required");
             }
 
+            if (!Guid.TryParse(orderId, out var id))
+                return BadRequest();
+
             var clientId = _requestContext.ClientId;
 
-            var activeOrders = await _limitOrdersRepository.GetActiveByClientIdAsync(clientId);
+            var order = await _historyClient.OrdersApi.GetOrderAsync(id);
 
-            if (activeOrders.All(x => x.Id != orderId))
+            if (order.WalletId != Guid.Parse(clientId))
```

Comment:
Are we expecting this request for Trading wallets only?

Reply:
Currently yes, maybe we will review this logic later in future tasks.